pinecone/index_connection_test.go

```diff
@@ -1073,3 +1135,139 @@ func TestToPaginationToken(t *testing.T) {
 		})
 	}
 }
+
+// Helper functions:
+func createVectorsForUpsert() []*Vector {
+	vectors := make([]*Vector, 5)
+	for i := 0; i < 5; i++ {
+		vectors[i] = &Vector{
+			Id:     fmt.Sprintf("vector-%d", i+1),
+			Values: []float32{float32(i), float32(i) + 0.1, float32(i) + 0.2, float32(i) + 0.3, float32(i) + 0.4},
+			SparseValues: &SparseValues{
+				Indices: []uint32{0, 1, 2, 3, 4},
+				Values:  []float32{float32(i), float32(i) + 0.1, float32(i) + 0.2, float32(i) + 0.3, float32(i) + 0.4},
+			},
+			Metadata: &structpb.Struct{
+				Fields: map[string]*structpb.Value{
+					"genre": {Kind: &structpb.Value_StringValue{StringValue: "classical"}},
+				},
+			},
+		}
+	}
+	return vectors
+}
+
+func setDimensionsForTestIndexes() uint32 {
+	return uint32(5)
+}
+
+func buildServerlessTestIndex(in *Client, idxName string) *Index {
+	ctx := context.Background()
+
+	fmt.Printf("Creating Serverless index: %s\n", idxName)
+	serverlessIdx, err := in.CreateServerlessIndex(ctx, &CreateServerlessIndexRequest{
+		Name:      idxName,
+		Dimension: int32(setDimensionsForTestIndexes()),
+		Metric:    Cosine,
+		Region:    "us-east-1",
+		Cloud:     "aws",
+	})
+	if err != nil {
+		log.Fatalf("Failed to create Serverless index \"%s\" in integration test: %v", err, idxName)
+	} else {
+		fmt.Printf("Successfully created a new Serverless index: %s!\n", idxName)
+	}
+	return serverlessIdx
+}
+
+func buildPodTestIndex(in *Client, name string) *Index {
+	ctx := context.Background()
+
+	fmt.Printf("Creating pod index: %s\n", name)
+	podIdx, err := in.CreatePodIndex(ctx, &CreatePodIndexRequest{
+		Name:        name,
+		Dimension:   int32(setDimensionsForTestIndexes()),
+		Metric:      Cosine,
+		Environment: "us-east-1-aws",
+		PodType:     "p1",
+	})
+	if err != nil {
+		log.Fatalf("Failed to create pod index in buildPodTestIndex test: %v", err)
+	} else {
+		fmt.Printf("Successfully created a new pod index: %s!\n", name)
+	}
+	return podIdx
+}
+
+func generateFloat32Array(n int) []float32 {
+	array := make([]float32, n)
+	for i := 0; i < n; i++ {
+		array[i] = float32(i)
```

Comment:
This is minor - would be nice to generate random floats or ints for these helpers rather than just using the same index values every time. I guess it ultimately doesn't matter much, but random values could be a nice stress test.

Reply:
Will do!

Reply:
I tried doing it by setting the start of the range to a random int, but because `n` is always so low (`5` or less), it often fails. I think we can just leave as is?

Reply:
We can leave as-is, it's nitpicky.

I think you could write a helper function that uses `math/rand` maybe:

```go
func RandomFloat32() float32 {
	rand.Seed(time.Now().UnixNano()) // Seed the random number generator
	return rand.Float32()
}
```